src/pybind/mgr/object_format.py

```diff
@@ -0,0 +1,272 @@
+# object_format.py provides types and functions for working with
+# requested output formats such as JSON, YAML, etc.
+
+import enum
+import errno
+import json
+
+from functools import wraps
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+try:
+    from typing import Protocol  # Protocol was added in Python 3.8
+except ImportError:
+
+    class Protocol:  # type: ignore
+        pass
+
+
+import yaml
+
+from mgr_module import HandlerFuncType
+
+
+class Format(enum.Enum):
+    plain = "plain"
+    json = "json"
+    json_pretty = "json-pretty"
+    yaml = "yaml"
+    xml_pretty = "xml-pretty"
+    xml = "xml"
+
+
+def _obj_to_json(obj: Any) -> Any:
+    tj = getattr(obj, "to_json", None)
+    if tj:
+        return tj()
+    return obj
+
+
+def _obj_to_yaml(obj: Any) -> Any:
+    if hasattr(obj, "yaml_representer"):
```

Comment:
For consistency:
```suggestion
    if hasattr(obj, "to_yaml"):
```

Reply:
Unfortunately, this would break existing code. When I first prototyped this idea I did my own approach to the serialization of the json/yaml but when I went back to better integrate it with existing code discovered that the orchestration module already had some general approaches to serialization. To avoid additional wheel-reinvention I refactored the existing orchestration code for additional reuse.  Renaming "yaml_representer" would break a bunch of existing code in the orchestration module(s).